features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/_RumInternalProxy.kt

```diff
@@ -31,6 +32,7 @@ import com.datadog.android.telemetry.model.TelemetryConfigurationEvent
     "VariableNaming"
 )
 class _RumInternalProxy internal constructor(private val rumMonitor: AdvancedRumMonitor) {
+    private var handledSyntheticsAttribute = false
```

Comment:
is this variable going to be handle from different threads ? In that case because you check for its value below the way I remember (hope I am not wrong) booleans are not volatile by default in JVM so you either use `@Volatile` annotation or use an `AtomicBoolean` in there. Also if this is the case you might need some unit tests that access these methods from multiple threads.

Reply:
Good observation. The new API should only be called internally, and only once. The variable prevents SDK consumers from calling it again and override the synthetics attribute. Since there is no guarantee that it won't be called from different threads, I believe it still makes sense to make it volatile or use an `AtomicBoolean`.